### DISCUSSION TITLE
Is there any way to get rid of Box<T> in the source?

### DISCUSSION THREAD

**Discussion by dmtrKovalenko** [pdeljanov/Symphonia#265] [General]
My use case is pretty simple — I need to decode a buffer while leaning toward a very transparent lifecycle of the top-level buffer. 

But with Symphonia I simply can't because you require the input to be Exactly the `Box` for some reason. Why? I am wondering if there is any way to receive a simple slice + Sync + Send and read all data directly?

**Comment by pdeljanov** [comment]
`MediaSourceStream` requires a boxed `dyn MediaSource` so that it can work for many possible media sources (e.g., `File`, `Cursor`, HTTP client, etc.) without having to be generic itself. The problem with making `MediaSourceStream` generic is that all `FormatReader`s would also have to be generic as well. For cases where you may have different media sources (e.g., a media player), this would lead to some extreme  binary bloating.

It sounds like your issue is related to lifetimes though? Would #236 help?

**Reply by dmtrKovalenko** [reply]
#236 will definitely help. It was something I considered to do myself until found that I can't read data back neither.
